doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,189 @@
+
+   See also `Functional.derivative` and `Operator.derivative`.
+
+
+ * ``functional.convex_conj``. This is the convex conjugate functional, also known as the Legendre transform or Fenchel conjugate.
+   It is defined as
+
+   .. math::
+
+      S^*(x^*) = \sup_{x \in X} \{ \langle x^*,x \rangle - S(x)  \},
+
+   where :math:`x^*` is an element in :math:`X` and :math:`\langle x^*,x \rangle` is the inner product.
+   (Note that :math:`x^*` should lives in the space :math:`X^*`, the (continuous/normed) `dual space`_ of :math:`X`, however since we assume that :math:`X` is a Hilbert space we have :math:`X^* = X`).
+
+ * ``proximal``. This returns a `proximal factory` for the proximal operator of the functional.
+   The proximal operator is defined as
+
+   .. math::
+
+      \text{prox}_{\sigma S}(x) = \text{arg min}_{y \in X} \{ S(y) + \frac{1}{2\sigma} \|y - x\|_2^2 \}. 
+
+The `Functional` class also contains default implementations of two help functions:
+
+ * ``derivative(point)``. Given an implementation of the gradient, this method returns the (directional) derivative operator in ``point``.
+   This is the linear operator
+
+   .. math::
+
+      x \to \langle x, \nabla S(point) \rangle,
+
+   where :math:`\nabla S(point)` is the gradient of the functional in the point :math:`point`.
+
+ * ``translated(shift)``. Given a functional :math:`S(.)`, this method creates the functional :math:`S(. - shift)`
+
+
+.. _dual space: https://en.wikipedia.org/wiki/Dual_space
+
+.. _calculus:
+
+Functional arithmetic
+=====================
+It is common in applications to perform arithmetic operations with functionals, for example adding two functionals :math:`S` and :math:`T`:
+
+.. math::
+   [S+T](x) = S(x) + T(x),
+
+or multiplication of a functional by a scalar:
+
+.. math::
+   [\alpha S](x) = \alpha S (x).
+
+Another example is translating a functional with a vector :math:`y`:
+
+.. math::
+   translate(S(x), y) = S(x - y),
+
+or given an `Operator` :math:`A` whose range is the same as the domain of the functional we also have composition:
+
+.. math::
+    [S * A](x) = S(A(x)). 
+
+In some of these cases, properties and methods such as ``gradient``, ``convex_conjugate`` and ``proximal`` can be calculated automatically given a default implementation of the corresponding property in :math:`S` and :math:`T`.
+
+All available functional arithmetic, including which properties and methods that automatically can be calculated, is shown below.
+``S``, ``T`` represent `Functional`'s with common domain and range, and ``A`` an `Operator` whose range is the same as the domain of the functional.
+``a`` is a scalar in the field of the domain of ``S`` and ``T``, and ``y`` is a vector in the domain of ``S`` and ``T``.
+
++--------------------+-----------------+--------------------------------------------------------------------------------+
+| Code               | Meaning         | Class                                                                          |
```

Comment:
This table makes the equivalent table for operators look like shit. I gotta put an eye at it at some point.